ui/reverse_proxy.go

```diff
@@ -107,6 +108,43 @@ func ReverseTequilapiProxy(tequilapiAddress string, tequilapiPort int, authentic
 	}
 }
 
+func parseToken(c *gin.Context) (string, error) {
+	// authenticate from header
+	token, err := parseHeaderToken(c)
+	if err != nil {
+		return "", err
+	}
+	if token != "" {
+		return token, nil
+	}
+
+	// authenticate from cookie
+	return parseHeaderToken(c)
+}
+
+func parseCookieToken(c *gin.Context) (string, error) {
+	token, err := c.Cookie(auth.JWTCookieName)
+	if err == http.ErrNoCookie {
+		// No error, just no token
```

Comment:
You're repeating the same comment twice here.
Also, if we return no error just an empty string if one occurs, can't we just be a little bit more "adventurous" and ignore the error completely? I mean we're not really handling it and if an errors occurs we'll get an empty string either way.

Reply:
Yes, if there is not error i return "".
But with `err` I want to recognise user input error here and return 400 in response